src/proxy.rs

```diff
@@ -873,41 +890,115 @@ fn is_cgi() -> bool {
     let proxy_enable: u32 = internet_setting.get_value("ProxyEnable")?;
     let proxy_server: String = internet_setting.get_value("ProxyServer")?;
 
-    Ok((proxy_enable, proxy_server))
+    if proxy_enable == 1 {
+        Some(proxy_server)
+    } else {
+        None
+    }
 }
 
-#[cfg(target_os = "windows")]
-fn get_from_registry() -> Option<RegistryProxyValues> {
-    get_from_registry_impl().ok()
-}
+#[cfg(target_os = "macos")]
+fn parse_setting_from_dynamic_store(
+    proxies_map: &CFDictionary<CFString, CFType>,
+    enabled_key: CFStringRef,
+    host_key: CFStringRef,
+    port_key: CFStringRef,
+    scheme: &str,
+) -> Option<String> {
+    let proxy_enabled = proxies_map
+        .find(enabled_key)
+        .and_then(|flag| flag.downcast::<CFNumber>())
+        .and_then(|flag| flag.to_i32())
+        .unwrap_or(0)
+        == 1;
+
+    if proxy_enabled {
+        let proxy_host = proxies_map
+            .find(host_key)
+            .and_then(|host| host.downcast::<CFString>())
+            .map(|host| host.to_string())
+            .unwrap_or_default();
+        let proxy_port = proxies_map
+            .find(port_key)
+            .and_then(|port| port.downcast::<CFNumber>())
+            .and_then(|port| port.to_i32());
+
+        if let Some(proxy_port) = proxy_port {
+            return Some(format_args!("{scheme}={proxy_host}:{proxy_port}").to_string());
+        } else {
+            return Some(format_args!("{scheme}={proxy_host}").to_string());
+        }
+    }
 
-#[cfg(not(target_os = "windows"))]
-fn get_from_registry() -> Option<RegistryProxyValues> {
     None
 }
 
-#[cfg(target_os = "windows")]
-fn parse_registry_values_impl(
-    registry_values: RegistryProxyValues,
-) -> Result<SystemProxyMap, Box<dyn Error>> {
-    let (proxy_enable, proxy_server) = registry_values;
+#[cfg(target_os = "macos")]
+fn get_from_platform_impl() -> Result<Option<String>, Box<dyn Error>> {
+    let store = SCDynamicStoreBuilder::new("reqwest").build();
+
+    if let Some(proxies_map) = store.get_proxies() {
+        let http_proxy_config = parse_setting_from_dynamic_store(
+            &proxies_map,
+            unsafe { kSCPropNetProxiesHTTPEnable },
+            unsafe { kSCPropNetProxiesHTTPProxy },
+            unsafe { kSCPropNetProxiesHTTPPort },
+            "http",
+        );
+        let https_proxy_config = parse_setting_from_dynamic_store(
+            &proxies_map,
+            unsafe { kSCPropNetProxiesHTTPSEnable },
+            unsafe { kSCPropNetProxiesHTTPSProxy },
+            unsafe { kSCPropNetProxiesHTTPSPort },
+            "https",
+        );
 
-    if proxy_enable == 0 {
-        return Ok(HashMap::new());
+        match (http_proxy_config, https_proxy_config) {
+            (Some(http_proxy_config), Some(https_proxy_config)) => {
+                return Ok(Some(
+                    format_args!("{http_proxy_config};{https_proxy_config}").to_string(),
```

Comment:
nit: use `format!` instead of `format_args!` (followed by a `to_string`) throughout

Reply:
Done 👍🏻